code/modules/food_and_drinks/kitchen_machinery/grill.dm

```diff
@@ -0,0 +1,110 @@
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/grill_fuel = 0
+	var/grilled_item
+	var/datum/looping_sound/grill/grill_loop
+
+/obj/machinery/grill/Initialize()
+	..()
+	grill_loop = new(list(src), FALSE)
+
+/obj/machinery/grill/update_icon()
+	if(grilled_item)
+		icon_state = "grill"
+	else if(!grill_fuel)
+		icon_state = "grill_open"
+	else
+		icon_state = "grill_on"
+
+/obj/machinery/grill/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/trash/coal))
+		qdel(I)
+		to_chat(user, "<span class='notice'>You put the [I] in [src].</span>")
+		add_fuel(500)
+		return
+	if(istype(I, /obj/item/stack/sheet/mineral/wood))
+		qdel(I)
+		to_chat(user, "<span class='notice'>You put the [S.amount] [I]s in [src].</span>")
+		var/obj/item/stack/S = I
+		add_fuel(50 * S.amount)
+		return
+	if(I.resistance_flags & INDESTRUCTIBLE)
+		to_chat(user, "<span class='warning'>You don't feel it would be wise to grill [I]...</span>")
+		return ..()
+	if(istype(I, /obj/item/reagent_containers))
+		if(istype(I, /obj/item/reagent_containers/food) && !istype(I, /obj/item/reagent_containers/food/drinks))
+			if(HAS_TRAIT(I, TRAIT_NODROP) || (I.item_flags & (ABSTRACT | DROPDEL)))
+				return ..()
+			else if(!grill_fuel)
+				to_chat(user, "<span class='notice'>There is not enough fuel.</span>")
+				return
+			else if(!grilled_item && user.transferItemToLoc(I, src))
+				grilled_item = I
+				to_chat(user, "<span class='notice'>You put the [I] on [src].</span>")
+				update_icon()
+				grill_loop.start()
+				return
+		else
+			add_fuel(20 * (I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy)))
+			to_chat(user, "<span class='notice'>You pour the Monkey Energy in [src].</span>")
+			I.reagents.remove_reagent(/datum/reagent/consumable/monkey_energy, I.reagents.get_reagent_amount(/datum/reagent/consumable/monkey_energy))
+			return
+	..()
+
+/obj/machinery/grill/proc/add_fuel(amount)
+	grill_fuel += amount
+	update_icon()
+
+/obj/machinery/grill/process()
+	..()
+	update_icon()
+	if(!grill_fuel)
+		return
+	else
+		grill_fuel -= 1
+		if(prob(1))
+			var/datum/effect_system/smoke_spread/bad/smoke = new
+			smoke.set_up(1, loc)
+			smoke.start()
+	if(grilled_item)
+		var/obj/item/reagent_containers/I = grilled_item
+		I.reagents.add_reagent(/datum/reagent/consumable/char, 1)
+		grill_fuel -= 10
+		I.AddComponent(/datum/component/sizzle)
+
+/obj/machinery/grill/wrench_act(mob/living/user, obj/item/I)
+	default_unfasten_wrench(user, I)
+	return TRUE
+
+/obj/machinery/grill/deconstruct(disassembled = TRUE)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		new /obj/item/stack/sheet/metal(loc, 5)
+		new /obj/item/stack/rods(loc, 5)
+	qdel(src)
+
+/obj/machinery/grill/attack_ai(mob/user)
+	return
+
+/obj/machinery/grill/attack_hand(mob/user)
+	if(grilled_item)
+		var/obj/item/reagent_containers/food/I = grilled_item
+		to_chat(user, "<span class='notice'>You take out [grilled_item] from [src].</span>")
+		I.forceMove(drop_location())
+		if(!issilicon(user))
```

Comment:
So what if they are a sillicon? Type check this way earlier and do an early return instead

Reply:
if they are a silicon it drops it on the ground instead of their hands??